File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        Map<String,List<String>> map = new HashMap<>();
        
        for(String[] list : clothes){
            String key = list[1];
            String value = list[0];
            
            if(map.get(key) == null){
                map.put(key, new ArrayList<>()); 
            }
            
            map.get(key).add(value);
               
        }
        
        for(Map.Entry<String, List<String>> entry : map.entrySet()){
            String key = entry.getKey();
            List<String> values = entry.getValue();
            answer*=values.size()+1;
            
        }
        
        answer-=1;
        
        return answer;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict
def solution(clothes):
    answer = 1
    hash_map=defaultdict(list)
    
    for value,key in clothes:
        if value not in hash_map[key]:
            hash_map[key].append(value)
    
    
    for li in hash_map.values():
        answer*=(len(li)+1)
    
    answer-=1
            
    
    
    return answer